### PR TITLE
Update Dockerfile

### PR DESCRIPTION
**Pull request by danbarr**
* Switch back to tfc-agent user.
* Remove entrypoint/cmd override, let single execution mode be controlled by the env var.
* Explicitly tag the source image (resolves Hadolint DL3006)

### CODE REVIEW

File: examples/tfc-agent-gke-simple/Dockerfile [hunk 1]
```diff
@@ -15,6 +16,4 @@ RUN mkdir -p /usr/local/gcloud \
 # Adding the package path to local
 ENV PATH $PATH:/usr/local/gcloud/google-cloud-sdk/bin
 
-# /home/tfc-agent/bin/tfc-agent is the entrypoint from the hashicorp image
-ENTRYPOINT ["/home/tfc-agent/bin/tfc-agent"]
```
Comment (gautambaghel) [inline examples/tfc-agent-gke-simple/Dockerfile:19]: @danbarr would we then specify the entry point command in the yaml file?

Comment (gautambaghel) [reply examples/tfc-agent-gke-simple/Dockerfile:19]: here? https://github.com/danbarr/terraform-google-terraform-cloud-agents/blob/fix-dockerfile/examples/tfc-agent-gke-simple/sample-manifests/deployment.yaml#L19

**Review by gautambaghel — CHANGES_REQUESTED** [review]

File: examples/tfc-agent-gke-simple/Dockerfile (hunk 1 above)
Comment (danbarr) [reply examples/tfc-agent-gke-simple/Dockerfile:19]: > @danbarr would we then specify the entry point command in the yaml file?

Shouldn't need to, it'll be inherited from the base container.

**Comment by gautambaghel** [comment]
@danbarr  - Ahh cool! Merging now

**Review by gautambaghel — APPROVED** [review]

**Pull request merged**